engine/resource/src/resource_mounts.cpp

```diff
@@ -482,7 +482,7 @@ static dmResource::Result GetDependenciesInternal(DependencyIterContext* ctx, co
 
         for (uint32_t d = 0; d < dependencies.Size(); ++d)
         {
-            dmhash_t dep_hash = dependencies[i];
+            dmhash_t dep_hash = dependencies[d];
```

Comment:
The initial issue with one type where iterator should use internal for-loop variable `d` instead of the variable `i` from an external for-loop